service.subtitles.subtitles.gr/resources/lib/subztvgr.py

```diff
@@ -0,0 +1,170 @@
+# -*- coding: utf-8 -*-
+
+'''
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+'''
+
+
+import urllib,urlparse,re,os
+
+from lamlib import cache
+from lamlib import cleantitle
+from lamlib import client
+from lamlib import control
+
+
+class subztvgr:
+    def __init__(self):
+        self.list = []
+
+
+    def get(self, query):
+        try:
+            match = re.findall('(.+?) \((\d{4})\)$', query)
+
+            if len(match) > 0:
+
+                title, year = match[0][0], match[0][1]
+
+                query = ' '.join(urllib.unquote_plus(re.sub('%\w\w', ' ', urllib.quote_plus(title))).split())
+
+                url = 'http://subztv.gr/search?q=%s' % urllib.quote_plus(query)
+
+                result = client.request(url)
+                result = re.sub(r'[^\x00-\x7F]+', ' ', result)
+
+                url = client.parseDOM(result, 'div', attrs = {'id': 'movies'})[0]
+                url = re.findall('(/movies/\d+)', url)
+                url = [x for y,x in enumerate(url) if x not in url[:y]]
+                url = [urlparse.urljoin('http://subztv.gr', i) for i in url]
+                url = url[:3]
+
+                for i in url:
+                    try:
```

Comment:
What is with all the nested try-except stuff? You're probably not solving a problem with these, but rather masking one.